File: An II/Sem I/MAP/Seminar/Sem1/src/Model/MessageTask.java
package Model;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MessageTask extends Task{
    private String message;
    private String from;
    private String to;
    private LocalDateTime date;

    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyy-MM-dd HH:mm");

    public MessageTask(String description, String id, String message, String from, String to, LocalDateTime date) {
        super(description, id);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }



    @Override
    public String toString() {
        return "id="
                + getID()
                + " | description="
                + getDescriere()
                + " | message="
                + message
                + " | from="
                + from
                + " | to="
                + to
                + " | date="
                + date.format(formatter);
    }

    @Override
    public void execute() {
        System.out.println(date.format(formatter) + ":" + message);
    }


}

File: An II/Sem I/MAP/Lab/Lab11/Lab11/Model/MessageTask.cs
namespace Lab11;

public class MessageTask : Task
{
    private String mesaj { get; set; }
    private String from { get; set; }
    private String to { get; set; }
    private DateTime date { get; set; }
    
    public MessageTask(string taskID, string descriere, string mesaj, string from, string to, DateTime date) : base(taskID, descriere)
    {
        this.mesaj = mesaj;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    public override void execute()
    {
        Console.WriteLine(this.mesaj + " " + this.date.ToString());
    }
}